tests/tcp.rs

```diff
@@ -1229,4 +1248,13 @@ fn tcp_no_events_after_deregister() {
     let mut buf = [0; 10];
     assert_eq!(stream.read(&mut buf).unwrap(), 4);
     assert_eq!(&buf[0..4], &[1, 2, 3, 4]);
+
+    stream2.write_all(&[1, 2, 3, 4]).unwrap();
+    expect_no_events(&mut poll, &mut events);
+
+    std::thread::sleep(std::time::Duration::from_millis(200));
```

Comment:
Why don't we wait for a readable event for stream?

Reply:
At this point the stream has benn deregitered, goal of the test is to check there are no events returned after deregister. But I understand what you're saying, and implemented expect_no_event_with_timeout instead of doing a plain sleep. I wanted to make sure there's stuff to read before calling read.

Reply:
`expect_no_events` already polls with a `50ms` timeout, so I think this explicit sleep following a call to `expect_no_events` may not be necessary--especially the fact that there is another call to it at the end of the test

Reply:
Could be, but I don't want the test result to depend on how fast expect_no_events is, or what timeout it uses internally. Doesn't hurt having an explicit timeout, it is safer imo.